version/0004/wiggum.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+# Wiggum
+# ======
+# A lightweight DevOps linter
+# Checking for consistent DevOps patterns across codebases
+
+set -e
+
+echo "Wiggum is running..."
+COLLECTED_ERRORS=()
+
+source .wiggum
+
+# root directory of the repository
+REPO_DIR="."
+# RELATIVE_PATH is the path to the infrastructure releated files.
+# default is the $REPO_DIR
+if [ -z "${RELATIVE_PATH}" ]; then
+  RELATIVE_PATH=$REPO_DIR
+fi
+
+# Check that a README file exists
+if [[ -z "${CHECK_README}" ]] || [[ "${CHECK_README}" == "true" ]];
+then
+  if [[ ! -f "${REPO_DIR}/README.md" ]] && [[ ! -f "${REPO_DIR}/README.txt" ]] && [[ ! -f "${REPO_DIR}/README" ]];
+  then
+    COLLECTED_ERRORS+=("README does not exist")
+  fi
+else
+  echo "Skipping README check"
+fi
+
+# Check that Docker files are present with preferred naming convention
+if [[ -z "${CHECK_DOCKER}" ]] || [[ "${CHECK_DOCKER}" == "true" ]];
+then
+  DOCKER_SHOULD_EXIST=( ${RELATIVE_PATH}/Dockerfile ${RELATIVE_PATH}/docker-compose.yml ${RELATIVE_PATH}/.dockerignore  )
+  for i in "${DOCKER_SHOULD_EXIST[@]}"
+  do
+    if [ ! -f "$i" ];
+    then
+      COLLECTED_ERRORS+=("Docker file $i does not exist")
+    fi
+  done
+
+  # Check that Dockerfiles are alpha sorted
+  if [ "$(ls ${RELATIVE_PATH}/*-Dockerfile 2> /dev/null | wc -l)" -ge "1" ];
+  then
+    COLLECTED_ERRORS+=("Dockerfile naming convention should be 'Dockerfile', 'Dockerfile.<env>' or 'Dockerfile-<env>'")
+  fi
+
+  # Check that there's a 'test' service in Docker Compose
+  if [[ -z "${CHECK_TESTS}" ]] || [[ "${CHECK_TESTS}" == "true" ]];
+  then
+    if [ "$(grep -c '^  test:' ${RELATIVE_PATH}/docker-compose.yml)" -eq "0" ];
+    then
+      COLLECTED_ERRORS+=("'test:' service not found in docker-compose.yml")
+    fi
+  fi
+
+  # Check that there's a .git entry in .dockerignore
+  if [ "$(grep -c '.git' ${RELATIVE_PATH}/.dockerignore)" -eq "0" ];
+  then
+    COLLECTED_ERRORS+=("'.git' not found in ./.dockerignore")
+  fi
+else
+  echo "Skipping Dockerfile check"
+fi
+
+
+
+# Check that Buildkite is present
+if [[ -z "${CHECK_BUILDKITE}" ]] || [[ "${CHECK_BUILDKITE}" == "true" ]];
+then
+  # set buildkite config path
+  BUILDKITE_DIR="${RELATIVE_PATH}/.buildkite"
+  BUILDKITE_FILE="pipeline.yml"
+
+  BUILDKITE_DIR_SHOULD_EXIST=( ${BUILDKITE_DIR} )
+  for i in "${BUILDKITE_DIR_SHOULD_EXIST[@]}"
+  do
+    if [ ! -d "$i" ];
+    then
+      COLLECTED_ERRORS+=("Buildkite directory $i does not exist")
+    fi
+  done
+  BUILDKITE_SHOULD_EXIST=( ${BUILDKITE_DIR}/${BUILDKITE_FILE} )
+  for i in "${BUILDKITE_SHOULD_EXIST[@]}"
+  do
+    if [ ! -f "$i" ];
+    then
+      COLLECTED_ERRORS+=("Buildkite file $i does not exist")
+    fi
+  done
+
+  if [[ -z "${CHECK_TESTS}" ]] || [[ "${CHECK_TESTS}" == "true" ]];
```

Comment:
presume you repeated this check here (rather than incorporating it into the previous IF check) for code segmentation and clarity? @tkhoa2711

Reply:
Yeah because each of these test check is nested inside another check (1 for docker and 1 for buildkite) so I think keeping them separate makes more sense.

E.g. if `CHECK_DOCKER=false`, `CHECK_BUILDKITE=true` and `CHECK_TESTS=true` then it won't verify tests for docker config but will do for buildkite.